apps/angular/16-master-dependency-injection/src/app/app.component.ts

```
@@ -22,9 +29,27 @@ export class ProductDirective {
 
-@Component({
-  imports: [TableComponent, CurrencyPipe, AsyncPipe, ProductDirective],
+@Directive({
+  selector: 'tr[currencyCode]',
   providers: [CurrencyService],
+})
+export class ProductRowDirective {
+  #currencyService = inject(CurrencyService);
+  currencyCode = input.required<string>();
+
+  #syncCodeEffectRef = effect(() => {
```

ESLint

[error] '`#syncCodeEffectRef`' is defined but never used.

(no-unused-private-class-members)